src/hooks/http.js

```diff
@@ -68,14 +68,35 @@ export const parseHttpRequestOptions = (options = {}, url) => {
     port,
     uri,
     host,
-    body: '', // XXX Filled by the httpRequestEndWrapper ( / Write)
+    body: '', // XXX Filled by the httpRequestEndWrapper or httpRequestEmitWrapper ( / Write)
     method,
     headers: lowerCaseObjectKeys(headers),
     protocol,
     sendTime,
   };
 };
 
+export const httpRequestEmitWrapper = requestData => emitFunc =>
+  function(eventName, arg) {
+    safeExecute(
+      () => {
+        if (arg && arg._httpMessage && arg._httpMessage._hasBody) {
+          const httpMessage = arg._httpMessage;
+          let lines = [];
+          if (httpMessage.hasOwnProperty('outputData')) {
```

Comment:
validate that you can run `httpMessage.outputData[0].data.split`, i.e.  that data is string, etc.

Reply:
It's wrapped in `safeExecute`